### DISCUSSION TITLE
Include "Embedding Inspector" in core functionality of SD.Next

### DISCUSSION THREAD

**Discussion by Threnos** [vladmandic/sdnext#3449] [Ideas]
There was a great extension called "Embedding inspector".
The core feature is to explore vector space of a model as well as create your own embeddings. The functionality so unique and idea simple that I wonder why it was not re-implemented to this date.
The original repo was not maintained for long and eventually deleted due to unknown reasons. But, to my great relief, the project survived due to forks. And after some trial and error I found one that actually works in SD.Next (although, only with backend: original): https://github.com/veyDer/embedding-inspector.

I ask to bring it in house (or, at least, make a fork and give it verification blip in Extensions tab) to not let the project die and encourage other people to explore vectors and create combinations to tailor their specific needs. On top of that, this feature will only add up to the value of already feature-rich SD.Next, bringing it more attention.

With it I was able to find new tags, synonyms, combine them, experiment with negatives impact on generations. After months of development, the two of my most successful creations [antiBlur](https://civitai.com/models/144775/antiblur) and [antiSketch](https://civitai.com/models/148339/antisketch) now have each over a thousand downloads and hundreds of likes.
Even if this pledge will not be fufilled, go on and install the extension manually (while it still works, hehe) and start doing great things!
And if anyone can point towards an alternative, please do so.

**Comment by vladmandic** [comment]
i don't want to bring in-house anything that is limited to backend=original.
now, if someone was to implement this to be compatible with diffusers, that's another story.
i'd love to see contributions like that!

**Reply by Threnos** [reply]
Thank you for the answer!
I was kinda expecting this limitation to be a determining factor, but figured it was worth the shot.
While it's no big deal for getting close with Python, If only I had a knowledge of inner workings of all that AI stuff...
